From d515cf3c603e695786695858d48a64d2f2e7d6aa Mon Sep 17 00:00:00 2001
From: Sergio Prado <sergio.prado@toradex.com>
Date: Tue, 12 Jul 2022 20:42:48 -0300
Subject: [PATCH] udisksdaemon: do not wait for polkit authority

We are building polkit without polkitd, which is responsible for
providing the polkit authority. Since polkit authority is absent,
udisksd waits for the default service_start_timeout limit of 25s and
delays the boot process. Avoid this delay by skipping the call to
polkit_authority_get_sync(). Also set daemon->authority to NULL, same
as what happens when polkit authority is not initialized.

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Sergio Prado <sergio.prado@toradex.com>
Signed-off-by: Abdur Rehman <abdur.rehman@toradex.com>
---
 src/udisksdaemon.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/udisksdaemon.c b/src/udisksdaemon.c
index 51e2bb5c57ca..5adc986f2f64 100644
--- a/src/udisksdaemon.c
+++ b/src/udisksdaemon.c
@@ -379,12 +379,10 @@ udisks_daemon_constructed (GObject *object)
   uuid_unparse (uuid, &uuid_buf[0]);
   daemon->uuid = g_strdup (uuid_buf);
 
-  daemon->authority = polkit_authority_get_sync (NULL, &error);
+  daemon->authority = NULL;
   if (daemon->authority == NULL)
     {
-      udisks_critical ("Error initializing polkit authority: %s (%s, %d)",
-                    error->message, g_quark_to_string (error->domain), error->code);
-      g_clear_error (&error);
+      udisks_warning ("Warning: Polkit authority cannot be initialized");
     }
 
   daemon->object_manager = g_dbus_object_manager_server_new ("/org/freedesktop/UDisks2");
-- 
2.25.1

